db/python/tables/project.py

```diff
@@ -1,51 +1,309 @@
+                user, project_id, readonly=readonly
+            )
+            if not can_use:
+                invalid_project_names.append(project_name)
+                continue
+
+            project_ids.append(project_id)
+
+        if invalid_project_names:
+            raise NoProjectAccess(invalid_project_names, readonly=readonly)
+
+        if len(project_ids) != len(project_names):
+            raise Exception(
+                'An internal error occurred when mapping project names to IDs'
+            )
+
+        return project_ids
+
+    async def check_project_creator_permissions(self, author):
+        """Check author has project_creator permissions"""
+        # check permissions in here
+        if self.allow_full_access:
+            return True
+
+        response = self._read_secret('sample-metadata', 'project-creator-users')
+        if author not in set(response.split(',')):
+            raise Forbidden(f'{author} does not have access to creating project')
+
+        return True
+
+    async def get_project_rows(
+        self, author: Optional[str] = None, check_permissions=True
+    ) -> List[ProjectRow]:
         """Get {name: id} project map"""
-        _query = 'SELECT name, id FROM project'
+        if check_permissions:
+            await self.check_project_creator_permissions(author)
+
+        _query = 'SELECT id, name, gcp_id, dataset, read_secret_name, write_secret_name FROM project'
         rows = await self.connection.fetch_all(_query)
-        return dict(rows)
+        return list(map(ProjectRow.from_db, rows))
+
+    async def create_project(
+        self,
+        project_name: str,
+        dataset_name: str,
+        gcp_project_id: str,
+        author: str,
+        read_secret_name: str,
+        write_secret_name: str,
+        create_test_project: bool,
+        check_permissions=True,
+    ):
+        """Create project row"""
+        if check_permissions:
+            await self.check_project_creator_permissions(author)
+
+        _query = """\
+INSERT INTO project (name, gcp_id, dataset, author, read_secret_name, write_secret_name)
+VALUES (:name, :gcp_id, :dataset, :author, :read_secret_name, :write_secret_name)
+RETURNING ID"""
+        values = {
+            'name': project_name,
+            'dataset': dataset_name,
+            'gcp_id': gcp_project_id,
+            'author': author,
+            'read_secret_name': read_secret_name,
+            'write_secret_name': write_secret_name,
+        }
+
+        project_id = await self.connection.fetch_val(_query, values)
+
+        if create_test_project:
+            values = {
+                'name': project_name + '-test',
+                'dataset': dataset_name,
+                'gcp_id': gcp_project_id,
+                'author': author,
+                'read_secret_name': read_secret_name.replace('-main-', 'test'),
+                'write_secret_name': write_secret_name.replace('-main-', 'test'),
```

Comment:
Perhaps `"test"` should be surrounded by dashes in these 2 lines too?
```suggestion
                'write_secret_name': write_secret_name.replace('-main-', '-test-'),
```